mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,383 @@
+            key_names.append(str(keys[ii]["key"]).replace("-", "b"))
+    end_times.append(keys[-1]["time"])
+    return KeyData(np.array([start_times, end_times]).astype(float).T, key_names)
+
+
+@io.coerce_to_string_io
+def load_key_music21(fhandle: TextIO, resolution=28):
+    """Load haydn op20 key data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+
+    Returns:
+        List[dict]: musical key data and relative time (offset (Music21Object.offset) * resolution).
+
+    """
+    return load_key_base(fhandle, resolution)
+
+
+@io.coerce_to_string_io
+def load_midi_path(fhandle: TextIO):
+    """Load path to midi file of haydn op20 musical piece
+
+    Args:
+        fhandle (str or file-like): path to score file
+
+    Returns:
+        str: midi file path
+
+    """
+    midi_path = os.path.splitext(fhandle.name)[0] + ".midi"
+    if not os.path.exists(midi_path):
+        score, _ = split_score_annotations(fhandle)
+        score.write("midi", fp=midi_path)
+    return midi_path
+
+
+@io.coerce_to_string_io
+def load_roman_numerals(fhandle: TextIO, resolution=28):
+    """Load haydn op20 roman numerals data from a file
+
+    Args:
+        fhandle (str or file-like): path to roman numeral annotations
+
+    Returns:
+        List[dict]: musical roman numerals data and relative time (offset (Music21Object.offset) * resolution)
+
+    """
+    _, rna = split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        figure = rn.figure
+        annotations.append({"time": time, "roman_numeral": figure})
+    return annotations
+
+
+def load_chords_base(fhandle: TextIO, resolution: int = 28):
+    """Load haydn op20 chords data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to chord annotations
+
+    Returns:
+        List[dict]: musical chords data and relative time (offset (Music21Object.offset) * resolution)
+
+    """
+    _, rna = split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        chord = rn.pitchedCommonName
+        annotations.append({"time": time, "chord": chord})
+    return annotations
+
+
+@io.coerce_to_string_io
+def load_chords(fhandle: TextIO, resolution: int = 28):
+    """Load haydn op20 chords data from a file
+    Args:
+        fhandle (str or file-like): path to chord annotations
+
+    Returns:
+        ChordData: chord annotations
+
+    """
+    chords = load_chords_base(fhandle, resolution)
+    start_times, end_times, chord_names = [0], [], [str(chords[0]["chord"])]
+    for ii, k in enumerate(chords):
+        if str(k["chord"]) != chord_names[-1]:
+            end_times.append(chords[ii]["time"] - 1)
+            start_times.append(chords[ii]["time"])
+            chord_names.append(str(chords[ii]["chord"]))
```

Comment:
It looks like these lines aren't tested - can you add a case that covers this in your tests?

Reply:
load_chords_base, load_chords and load_chords_music21 are tested in the same test function: test_load_chords

Reply:
yes, the overall functions are tested, but these lines (the when the `if` case is true) are not tested

Reply:
You can see untested lines here: https://github.com/mir-dataset-loaders/mirdata/pull/447/checks?check_run_id=2369338726

Reply:
ok! thanks!

Reply:
done

Reply:
Done!